clang/lib/Parse/Parser.cpp

```diff
@@ -1255,8 +1256,12 @@ Parser::DeclGroupPtrTy Parser::ParseDeclarationOrFunctionDefinition(
   // Add an enclosing time trace scope for a bunch of small scopes with
   // "EvaluateAsConstExpr".
   llvm::TimeTraceScope TimeScope("ParseDeclarationOrFunctionDefinition", [&]() {
-    return Tok.getLocation().printToString(
-        Actions.getASTContext().getSourceManager());
+    llvm::TimeTraceMetadata M;
+    const SourceManager &SM = Actions.getASTContext().getSourceManager();
+    auto Loc = SM.getExpansionLoc(Tok.getLocation());
```

Comment:
Could we extract the function that fills `SourceLocation` into `TimeTraceMetadata`?
This can be done in a follow up, but we are starting to have too many places now that do the same thing and should all be changed together (e.g. if we'd want to spell filename in some other way or if we'd want to use something other than expansionLocation in the future, or we'd want to add a column, etc)

Reply:
I went ahead and did this PR.
I do not have a great home for such a function, went ahead with SourceManager.